doc/build/sysbuild/index.rst

```diff
@@ -648,3 +693,285 @@ each image that is part of a project. Alternatively, there are
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     my_group
+     EXCLUDE hello_world
+   )
+
+Similarly, you can filter out entire groups of images:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     another_group
+     INCLUDE_FROM my_group
+     EXCLUDE_FROM other_group
+   )
+
+This example defines ``another_group``, which contains images that are unique to
+``my_group`` and do not overlap with ``other_group``. This illustrates a basic
+set difference operation for sysbuild groups.
+
+Exclusion takes precedence over inclusion when evaluating the contents of each
+group. For example, recall that the image ``my_sample`` has been added to both
+``my_group`` and ``other_group``, but it is filtered out from ``another_group``.
+Furthermore, trying to add that image back to the group by naïvely extending it:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     another_group
+     INCLUDE my_sample
+   )
+
+would not work, because the prior ``EXCLUDE_FROM`` directive still applies with
+higher priority. If you do need this to work, then you should consider defining
+yet another group:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     yet_another_group
+     INCLUDE_FROM another_group
+     INCLUDE my_sample
+   )
+
+In this way, ``yet_another_group`` can successfully include ``my_sample``,
+even though ``another_group`` excludes it. This is because group contents are
+evaluated in isolation, so by the time ``another_group`` is merged in, its own
+inclusions and exclusions will already have been squashed. The same goes for
+``my_group`` and ``other_group``, which need to be resolved like this first.
+
+Adding dependencies among sysbuild groups
+=========================================
+
+Just like for individual Zephyr applications, you can add configuration or
+flashing dependencies for groups of these applications.
+See :ref:`sysbuild_zephyr_application_dependencies` for more information.
+
+Here is an example of group-wise dependencies:
+
+.. code-block:: cmake
+
+   sysbuild_add_dependencies(GROUP CONFIGURE my_group group_a group_b)
+   sysbuild_add_dependencies(GROUP FLASH     my_group group_a group_b)
+
+This will ensure that every image in ``group_a`` and ``group_b`` will be
+configured and flashed before every image in ``my_group``. The ordering within
+the groups themselves is not determined here. Moreover, groups are not treated
+as units; once other image-wise and group-wise dependencies are factored in,
+the images belonging to one group are not expected to stick together in the
+sorted order.
+
+Note that ``my_group`` is allowed to overlap with ``group_a`` or ``group_b``.
+If two groups overlap and one group depends on the other, then the images that
+the two groups have in common will be placed in the middle of the sorted order,
+relative to the images unique to either group.
+
+To help illustrate what this means, consider a different example:
```

Comment:
although the below example is correct, then I cannot relate the text `then the images that the two groups have in common will be placed in the middle of the sorted order`, as to why there is no:
```
sysbuild_add_dependencies(IMAGE FLASH i_1 i_2 i_3)
sysbuild_add_dependencies(IMAGE FLASH i_2 i_1 i_3)
sysbuild_add_dependencies(IMAGE FLASH i_3 i_1 i_2)
```
cause group_a contains also i_1, i_2, i_3, but so does group_b.

I know why it is like this, but the text with current example must be much clearer, and maybe it's better with more groups than more images per each group, as I believe there probably will be created more groups but with just a few images in each (except for the `all` group).

Reply:
Agreed, I think I got a bit lazy in this section. I'll rewrite it.